geoip2_extras/middleware.py

```diff
@@ -70,28 +100,38 @@ def remote_addr(self, request):
         # http://stackoverflow.com/a/37061471/45698
         return header.split(',')[-1]
 
+    def get_geo_data(self, ip_address):
+        """Return City and / or Country data for an IP address."""
+        return self.city(ip_address) if self.geoip2._city else self.country(ip_address)
```

Comment:
If the GeoIP2() object has something in its `_city` attribute, then we have access to the City database. If not, we default to the Country database. There _should_ always be one or other (or both) of them because the `__init__` method will fail and disable the middleware if neither exists.